### DISCUSSION TITLE
taskReferenceName needs to be unique within a workflow

### DISCUSSION THREAD

**Discussion by shahnaresh41** [Netflix/conductor#2662] [Q&A] [answered]
Moving from conductor-2.30.0 to 3.3.4 and trying to import old workflow definition to conductor-3.3.4 
but it is throwing error that taskReferenceName needs to be unique within a workflow. 
Is there any solution to this so that we do not have to change the taskReferenceNamewithin workflow.

Error :
------------------
{
    "status": 400,
    "message": "Validation failed, check below errors for detail.",
    "retryable": false,
	"validationErrors": [
			{
				"path": "updateWorkflowDef.workflowDefList[0]",
				"message": "taskReferenceName: ABC_taskReferenceName1 should be unique across tasks for a given workflowDefinition: MyWorkflow"
			}
	]	
}

**Comment by apanicker-nflx** [comment, marked as answer]
The constraint that taskReferenceName needs to be unique in a given workflow was always implicit within Conductor, but was only enforced in [v2.5.0 ](https://github.com/Netflix/conductor/tree/v2.5.0/core/src/main/java/com/netflix/conductor/validations). 

This was a necessary validation because task reference names are used to wire inputs and outputs of tasks within the workflow. If the task reference names are not unique, then the JSON parser will not be able to correctly evaluate these expressions/wiring payloads between tasks.
